Initial commit

jeremykubica opened the pull request:
Create an example program to demonstrate one approach for computing which geocentric pointings overlap a single proposed heliocentric pointing and distance.

This code is currently meant as an example and is not optimized for speed or accuracy.

jeremykubica committed:
Initial version of the code

jeremykubica committed:
Update README.md

jeremykubica committed:
apply isort

jeremykubica committed:
apply isort

jeremykubica committed:
Update README.md

jeremykubica committed:
Update conftest.py

jeremykubica committed:
Add astropy and PPT badges

maxwest-uw commented on src/pointings_search/geometry.py:
```diff
@@ -0,0 +1,80 @@
+    return ra, dec
+
+
+def angular_distance(pts1, pts2):
```
this is the formula for angular distance between two normal vectors right? we should specify that so nobody tries shove ra/dec in there 😭 

also generally I would default to using astropy's distance calculation because they cover a lot of special cases, specifically at the poles, although there's is very slow so I understand not wanting to use it. and also the ecliptic doesn't pass through the poles afaik. do we have a sense of scale for the amount of points that we plan on running through this?

jeremykubica committed:
Update geometry.py

maxwest-uw commented on src/pointings_search/pointing_table.py:
```diff
@@ -0,0 +1,303 @@
+        if "sun_vec" not in self.pointings.columns:
+            self.append_sun_pos()
+        if "unit_vec" not in self.pointings.columns:
+            self.append_unit_vector(self)
```
another small thing, but is there a reason why you're passing self here to a function called from self?

jeremykubica committed:
Redo everything

Redo everything to:
1) Stay in ICRS frames
2) Use only astropy functions where possible

jeremykubica committed:
Fix isort problems

jeremykubica committed:
Fix isort problem

jeremykubica committed:
Add more tests

jeremykubica committed:
Fix isort problems

jeremykubica committed:
Lots of updates

jeremykubica committed:
Fix isort error

jeremykubica committed:
Add a demo notebook

jeremykubica committed:
Remove notebooks from documentation.

jeremykubica committed:
Fix broken clearing notebook output

jeremykubica committed:
Fix broken isort

jeremykubica committed:
Fix formatting errors
